src/openzaak/components/catalogi/admin/admin_views.py

```diff
@@ -148,28 +154,62 @@ def get(self, request, *args, **kwargs):
         return super().get(request, *args, **kwargs)
 
     def post(self, request, *args, **kwargs):
+
+        catalogus_pk = kwargs.get("catalogus_pk")
+        catalogus = Catalogus.objects.get(pk=catalogus_pk)
+
+        if "iotype-TOTAL_FORMS" in request.POST:
+            iotype_forms = InformatieObjectTypeFormSet(
```

Comment:
This comment is optional but I think it would be more readable to refactor this view and to initialize formsets in one place with parameters depending on if it's POST of GET. Like django does in it's `FormView` and its `get_form_kwargs` method. It will make code DRYer and will reduce a good amount of if-else from `post` method.

Reply:
Moved everything into the `get_context_data` funciton